### PR TITLE
Autosave preferences can be read after Nuke opens the script

### PR DESCRIPTION
**Pull request by jrsndl**
## Changelog Description
Looks like I need to open the script in Nuke to be able to correctly load the autosave preferences.
This PR reads the Nuke script in context, and offers owerwriting the current script with autosaved one if autosave exists.

## Testing notes:
1. Launch Nuke via Openpype
2. Go to Openpype menu and click work file
3. Open any nuke script with autosave
4. It will pop up nuke prompt to ask if you want to open autosave script.
5. It will load autosave script for you if you click yes; while it wont if you click no.

Also try to change autosave location in Nuke prefs to some random place, wait till autosave kicks in and kill Nuke.
OP should detect and offer autosaves that is not located next to the workfile as well.

### CODE REVIEW

**Comment by 64qam** [comment]
I think this has been resolved here https://github.com/ynput/OpenPype/pull/5107

**Review by jakubjezek001 — CHANGES_REQUESTED** [review]
Need to fix the nuke version condition to be from `14` perhaps?

File: openpype/hosts/nuke/api/workio.py [hunk 1]
```diff
@@ -21,21 +22,40 @@ def save_file(filepath):
 
 
 def open_file(filepath):
+
+    def read_script(nuke_script):
+        nuke.scriptClear()
+        if int(nuke.NUKE_VERSION_MAJOR) > 12:
```
[the remark this answers is not in the document]
Comment (jrsndl) [reply openpype/hosts/nuke/api/workio.py:28]: I didn't test with Nuke 13.x.
I just noticed that in Nuke 12, using nuke.scriptReadFile causes Nuke to "not know the nuke.root.name", but using nuke.scriptOpen is fine.

Maybe the error you are getting would happen anytime you open script from Higher version?

I will test Nuke 13

**Comment by jrsndl** [comment]
I 've removed the Nuke 12 compatibility condition for now. It is too old anyway.

**Review by jakubjezek001 — APPROVED** [review]

**Pull request merged**